### ISSUE TITLE
[FEATURE] Save Configuration

### ISSUE THREAD

**Issue by thunderclap82** [benbusby/whoogle-search#838]
It would be nice if all Whoogle's configurations were saved. Right now only Dark Theme, Root URL and Custom CSS saves, however I would like other things like Country, Near City Name, and Replace Social Media Links disabled. Thanks.

**Comment by liberninja** [comment]
I've been trying to figure this out as well. The documentation mentions the whoogle.env file to add permanent settings, but I can't for the life of me figure out how to access it to edit it.

**Comment by thunderclap82** [comment]
In the doc, I see you can add a variable of WHOOGLE_CONFIG_ALTS (I assume 0=new, 1=yes) but that doesn't seem to work as safe search constantly re-ticks.

**Comment by liberninja** [comment]
I still haven't figured out how to properly setup the WHOOGLE_DOTENV=1 variable. No matter what I do no custom settings are loaded...

**Comment by jacr13** [comment]
> I'm using Whoogle in docker and in the doc, I see you can add a variable of WHOOGLE_CONFIG_ALTS (I assume 0=no, 1=yes) but that doesn't seem to work as safe search constantly re-ticks.

try to update to the most recent version and adding WHOOGLE_CONFIG_SAFE=0 to stop using safe search.

**Comment by liberninja** [comment]
Yes, I renamed it to whoogle.env, configured it the way I want, and launched it with WHOOGLE_DOTENV=1
Still not applying the changes though.

**Comment by jacr13** [comment]
Which version are you using? Which deployment (with which command)? Do you mind sharing your whoogle.env?

**Comment by thunderclap82** [comment]
> > I'm using Whoogle in docker and in the doc, I see you can add a variable of WHOOGLE_CONFIG_ALTS (I assume 0=no, 1=yes) but that doesn't seem to work as safe search constantly re-ticks.
> 
> try to update to the most recent version and adding WHOOGLE_CONFIG_SAFE=0 to stop using safe search.

Safe is actually staying unchecked. The one I really want to disable is Replace Social Media Links. I want that off which looks like the variable is WHOOGLE_CONFIG_ALTS, however it's not staying unchecked. Every few days it returns.

**Comment by jacr13** [comment]
Whoogle has now the possibility to encode preferences in URL, it is maybe a solution for your problem

**Comment by liberninja** [comment]
> Whoogle has now the possibility to encode preferences in URL, it is maybe a solution for your problem

Oh cool! The main thing I need is the 'full size image search' option. Can that be used in the url? I'm not seeing any info about it in the documentation.

**Comment by liberninja** [comment]
> Which version are you using? Which deployment (with which command)? Do you mind sharing your whoogle.env?

I was using the 'Manual' setup before, but now I'm running it on docker. Still no custom config though. Here's the command I'm using to create the container:

`sudo docker run --publish 5001:5000 --detach --name whoogle-search benbusby/whoogle-search:latest --env-file ./whoogle.env`

which gives me this error:
```
4f47efe546484bf81986f9aa25f87cc65957be64887271220e35648c6390d89d
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "--env-file": executable file not found in $PATH: unknown.
```

Am I supposed to manually create a whoogle.env file somewhere?

**Comment by liberninja** [comment]
Oh well I'll be darned, it works now that I reinstall it! Must have put something in wrong before. I appreciate the help 👍
